src/controller.rs

```diff
@@ -23,101 +27,157 @@ struct Context {
     client: Client,
 }
 
-async fn reconcile(sinker: Arc<ResourceSync>, ctx: Arc<Context>) -> Result<Action> {
-    let client = ctx.client.clone();
+#[cfg(deleteme)]
+async fn resource_fetcher(
+    resource_ref: &ClusterResourceRef,
+    local_ns: &str,
+    ctx: Arc<Context>,
+) -> Result<()> {
+    let client = cluster_client(resource_ref.cluster.as_ref(), local_ns, ctx).await?;
+    Ok(())
+}
+
+async fn cluster_client(
+    cluster_ref: Option<&ClusterRef>,
+    local_ns: &str,
+    ctx: Arc<Context>,
+) -> Result<Client> {
+    let client = match cluster_ref {
+        None => ctx.client.clone(),
+        Some(cluster_ref) => {
+            let secrets: Api<Secret> = Api::namespaced(ctx.client.clone(), local_ns);
+            let secret_ref = &cluster_ref.kube_config.secret_ref;
+            let sec = secrets.get(&secret_ref.name).await?;
+
+            let kube_config = kube::config::Kubeconfig::from_yaml(
+                std::str::from_utf8(&sec.data.unwrap().get(&secret_ref.key).unwrap().0)
+                    .map_err(Error::KubeconfigUtf8Error)?,
+            )?;
+            let mut config =
+                Config::from_custom_kubeconfig(kube_config, &Default::default()).await?;
+
+            if let Some(ref namespace) = cluster_ref.namespace {
+                config.default_namespace = namespace.clone();
+            }
+
+            debug!(?config.cluster_url, "connecting to remote cluster");
+            let remote_client = kube::Client::try_from(config)?;
+            let version = remote_client.apiserver_version().await?;
+            debug!(?version, "remote cluster version");
+
+            remote_client
+        }
+    };
+    Ok(client)
+}
 
+async fn api_for(
+    cluster_resource_ref: &ClusterResourceRef,
+    local_ns: &str,
+    ctx: Arc<Context>,
+) -> Result<(Api<DynamicObject>, ApiResource)> {
```

Comment:
i think it might be nicer to have a struct here rather than a tuple but not a deal breaker

Reply:
I'm modeling this on an existing kube-rs API I'm just providing a convenience wrapper